runtime/sema/access.go

```diff
@@ -74,48 +74,93 @@ func NewEntitlementSetAccess(
 
 func (EntitlementSetAccess) isAccess() {}
 
-func (a EntitlementSetAccess) Description() string {
-	return a.AccessKeyword()
-}
+func (e EntitlementSetAccess) ID() TypeID {
+	entitlementTypeIDs := make([]TypeID, 0, e.Entitlements.Len())
+	e.Entitlements.Foreach(func(entitlement *EntitlementType, _ struct{}) {
+		entitlementTypeIDs = append(
+			entitlementTypeIDs,
+			entitlement.ID(),
+		)
+	})
 
-func (a EntitlementSetAccess) AccessKeyword() string {
-	return a.string(func(ty Type) string { return ty.QualifiedString() })
+	// FormatEntitlementSetTypeID sorts
+	return FormatEntitlementSetTypeID(entitlementTypeIDs, e.SetKind)
 }
 
-func (a EntitlementSetAccess) AuthKeyword() string {
-	return fmt.Sprintf("auth(%s)", a.AccessKeyword())
+func FormatEntitlementSetTypeID[T ~string](entitlementTypeIDs []T, kind EntitlementSetKind) T {
+	var builder strings.Builder
+	var separator string
+
+	switch kind {
+	case Conjunction:
+		separator = ","
+	case Disjunction:
+		separator = "|"
```

Comment:
Should maybe keep a space after the separator?

Reply:
We do not need and should not keep any spaces in type IDs, given that space usage matters (encoding in execution state).